core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -3151,9 +3152,31 @@ private[spark] object Utils
   }
 }
 
-private[util] object CallerContext extends Logging {
-  val callerContextEnabled: Boolean =
-    SparkHadoopUtil.get.conf.getBoolean("hadoop.caller.context.enabled", false)
+private[spark] object CallerContext extends Logging {
+  var callerContextEnabled: Boolean = SparkHadoopUtil.get.conf.getBoolean(
+    HADOOP_CALLER_CONTEXT_ENABLED_KEY, HADOOP_CALLER_CONTEXT_ENABLED_DEFAULT)
+
+  /**
+   * Run a function with the caller context enabled flag overridden to the given value. Normally,
+   * the flag is initialized from Hadoop configuration once per JVM process and never changed. This
+   * function is intended only for use by tests that specifically need to control whether the caller
+   * context is enabled. The configured value will be restored after execution of the function.
+   *
+   * @param enabled override value for the caller context enabled flag
+   * @param func function to run
+   *
+   * VisibleForTesting
+   */
+  def withCallerContextEnabled[T](enabled: Boolean)(func: => T): T = {
```

Comment:
This looks like a pure test-helper utility instead of `VisibleForTesting`. Do we have any benefit in `main` code?

Reply:
There is no benefit to main code, other than keeping all access to `callerContextEnabled` in the same object. Otherwise, people reading the code might be confused about why it's `var` instead of `val`.

If you prefer, I can just leave a comment on why it's `var` and move this to a new `CallerContextTestUtils` object under test (or some existing test utils file if you have another suggestion).

I would potentially like to reuse this in tests covering other usage of caller context, which is why I didn't put it directly in `FsHistoryProviderSuite`.